src/index/import/markdown.ts

```diff
@@ -381,3 +410,43 @@ export function parseFrontmatter(value: any): Literal {
     // Backup if we don't understand the type.
     return null;
 }
+/** recursively parse yaml links and return them as a single array */
+export function parseYAMLLinks(obj: any): Link[] {
+    let links: Link[] = [];
+    for (let v in obj) {
+        let obj2 = parseFrontmatter(obj[v]);
+
+        if (typeof obj2 === "object") {
+            if (Array.isArray(obj2)) {
+                let result = [];
+                for (let child of obj2 as Array<any>) {
+                    result.push(parseFrontmatter(child));
+                }
+                for (let c of result) {
+                    if (c instanceof Link) addLink(links, c);
+                    else {
+                        let inner = parseYAMLLinks(c);
+                        for (let n of inner) addLink(links, n);
+                    }
+                }
+            } else {
+                let object = obj2 as Record<string, any>;
+                let result: Record<string, Literal> = {};
+                for (let key in object) {
+                    result[key] = parseFrontmatter(object[key]);
+                    let rk = result[key];
+                    if (rk instanceof Link) {
+                        addLink(links, rk);
+                    } else {
+                        let inner = parseYAMLLinks(rk);
+                        for (let n of inner) addLink(links, n);
+                    }
+                }
+            }
+        } else if (typeof obj2 === "string") {
+            let linkParse = EXPRESSION.link.parse(obj2);
+            if (linkParse.status && linkParse.value.value instanceof Link) addLink(links, linkParse.value.value);
```

Comment:
You don't need to check instanceof Link, if the parse succeeds it will always return a `Link`.